apps/lockfile-explorer-web/src/components/ConnectionModal/index.tsx

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import React, { useCallback, useEffect, useState } from 'react';
+import styles from './styles.scss';
+import appStyles from '../../App.scss';
+import { checkAlive } from '../../parsing/getPackageFiles';
+import { ReactNull } from '../../types/ReactNull';
+
+export const ConnectionModal = (): JSX.Element | ReactNull => {
+  const [isAlive, setIsAlive] = useState(true);
+  const [checking, setChecking] = useState(false);
+  const [manualChecked, setManualChecked] = useState(false);
+
+  async function keepAlive(): Promise<void> {
+    if (await checkAlive()) {
+      setIsAlive(true);
+    } else {
+      setIsAlive(false);
+    }
+    setChecking(false);
+  }
+
+  useEffect(() => {
+    window.setInterval(keepAlive, 2000);
+  }, []);
+
+  const checkAliveManual = useCallback(() => {
+    setChecking(true);
+    setManualChecked(true);
+    keepAlive().catch((e) => {
```

Comment:
Just suppress the ESLint rule.

Reply:
@iclanton I disagree with this practice.  Perhaps `keepAlive()` cannot fail today, but in the future some person may come along and modify that function, unaware of the assumption here that it's not supposed to throw.  In its current form already, to prove that `keepAlive()` cannot fail we must inspect an external source file to analyze the `checkAlive()` function as well.

The safest practice is to add the catch block that logs if the assumption is violated (e.g. `console.error('Unexpected exception: ' + e.toString());`).  The savings of omitting such safeguards really aren't justified.

I would go further to say that `no-floating-promises` lint rule is one that should /never/ be suppressed.  A couple weeks ago I met a senior engineer for coffee to help him debug a mysterious promise failure, whose ultimate cause was that someone else had mistakenly disabled this lint rule in one file. 😆

Reply:
Added the console.error message